### ISSUE TITLE
Suggestion

### ISSUE THREAD

**Issue by yakir12** [JuliaData/MarkdownTables.jl#1]
If this is supposed to be **the** place for markdown tables, then check [this](https://github.com/yakir12/MDtables.jl) out. More explanations can be found in [this](https://discourse.julialang.org/t/markdown-tables/11385/21?u=yakir12) Discourse post.

**Comment by nalimilan** [comment]
I'm really not sure this kind of package should live in JuliaData. Maybe let's just delete this repo?

**Comment by yakir12** [comment]
Seems like a placeholder, just thought I'd ping in my efforts about this kind of thing.

**Comment by quinnj** [comment]
Yes, we can delete this repo and you can have the name if you'd like. I'd also be open to moving your package into JuliaData if you'd like. The idea here was that we'd have HTMLTables.jl, MarkdownTables.jl, etc. that all support the Tables.jl interface.

**Comment by yakir12** [comment]
That sounds fun. Any simple example of a package that has adopted the Tables.jl interface, so I can see what it takes in the wild?

**Comment by quinnj** [comment]
Here's the DataFrames example: https://github.com/JuliaData/DataFrames.jl/blob/master/src/other/tables.jl
And here's CSV.jl: https://github.com/JuliaData/CSV.jl/blob/master/src/tables.jl
Let me know if you have any questions!

**Comment by yakir12** [comment]
Had a moment to look at this, I'm confused: all my `MDtables.jl` does is prints an abstract array in markdown format. I don't understand how `Tables.jl`'s row/column API applies here..? For instance, there is no reason for someone to save their tabular data as a markdown table "in memory". The markdown part comes into the picture only when printing tabular data (and reading it, which I have not yet implemented). But up until that point, the user can keep her data in whatever type she likes (DateFrame, AbstractArray, etc).

**Comment by quinnj** [comment]
My main thought for MDtables.jl is that it could provide a:
```julia
function markdowntable(x)
    sch = Tables.schema(x)
    # code to print table header w/ schema
    for row in Tables.rows(x)
        # code to print each row in markdown format
    end
    return
end
```
So yeah, there's not really a use-case for MDtables.jl as a tables _source_, but mainly as a _sink_ where people might want to format arbitrary sources as markdown table outputs.

**Comment by o314** [comment]
There is also the Markdown package in the julia stdlib @ https://github.com/JuliaLang/julia/blob/v1.0.3/stdlib/Markdown/src/GitHub/table.jl

--
Answering multiple remarks there :

Markdown could go to ascii, terminal or ... markdown at least.

If a separated package seems too slim, one could either

- contribute back to the near (and lean, please) one :
    - markdown @ stdlib
    - <https://github.com/JuliaData/DataTables.jl> - seems to be the next AbstractTable on the block
    - <https://github.com/JuliaComputing/IndexedTables.jl> - at the core of juliadb
- extend format supports in the vein of prettytable or pandoc (which handle 4 ascii table output)

**Comment by nalimilan** [comment]
DataTables is deprecated (as written in the project description)... and IndexedTables seems totally unrelated to Markdown tables.

**Comment by o314** [comment]
1. point - agree, link corrected. thanks.
2. point - totally wrong and completed related. as long as we need to render table in markdown, every table is concerned. in particular the most used.
